File: 8-kyu/cat-years,-dog-years/python/solution.py
def human_years_cat_years_dog_years(human_years):
    pet_years = lambda human_yrs, factor: 15 * (human_yrs > 0) + \
                                           9 * (human_yrs > 1) + \
                                           (human_yrs - 2) * factor * (human_yrs > 2)
    CAT_FACTOR = 4
    DOG_FACTOR = 5
    cat_years = pet_years(human_years, CAT_FACTOR)
    dog_years = pet_years(human_years, DOG_FACTOR)
    return [human_years, cat_years, dog_years]

File: 8-kyu/cat-years,-dog-years/typescript/solution.ts
const petYears = (humanYears: number, factor: number): number => {
  return 15 * Number(humanYears >= 1) +
    9 * Number(humanYears >= 2) +
    (humanYears - 2) * factor * Number(humanYears >= 3);
}

export const humanYearsCatYearsDogYears = (humanYears: number): [number, number, number] => {
  const dogFactor = 5;
  const catFactor = 4;
  const dogYears = petYears(humanYears, dogFactor);
  const catYears = petYears(humanYears, catFactor);
  return [humanYears, catYears, dogYears];
};

